activerecord/lib/active_record/autosave_association.rb

```diff
@@ -274,6 +283,17 @@ def changed_for_autosave?
     end
 
     private
+      # Track if this record is currently being saved.
+      # Autosave can call save multiple times on the same record. Some methods
+      # like +changes_applied+ should be called only once while saving.
+      def _saving
+        previously_saving, @_saving = @_saving, true
+        yield
+      ensure
+        @_saving = previously_saving
+        @_already_called[:changes_applied] = false unless @_saving
```

Comment:
Setting `@_already_called[:method_name]` to false is what is used in other places in this file.
But maybe it's better to just delete it:
```suggestion
        @_already_called(:changes_applied) unless @_saving
```